framework/include/cppmicroservices/detail/BundleAbstractTracked.tpp

```diff
@@ -110,18 +112,21 @@ void BundleAbstractTracked<S,TTT,R>::Close()
 template<class S, class TTT, class R>
 void BundleAbstractTracked<S,TTT,R>::Track(S item, R related)
 {
-  std::shared_ptr<TrackedParamType> object;
+  bool isInMap = false;
+  TrackedParamType object;
   {
     auto l = this->Lock(); US_UNUSED(l);
-    if (closed)
+    if (closed) 
     {
       return;
     }
     auto trackedItemIter = tracked.find(item);
-    if (trackedItemIter != tracked.end()) {
+    if (trackedItemIter != tracked.end()) 
+    {
       object = trackedItemIter->second;
+      isInMap = true;
     }
-    if (!object)
+    if (!isInMap) 
```

Comment:
They are split since the first occurs in a locked region (manages map) and the second must not (calls customizers). Is there a better way to write this?